querybook/server/lib/query_analysis/templating.py

```diff
@@ -168,28 +186,38 @@ def _flatten_variable(
     variable_defs: Dict[str, str],
     variables_dag: Dict[str, Set[str]],
     flattened_variables: Dict[str, str],
+    jinja_env,
 ):
     """ Helper function for flatten_recursive_variables.
         Recursively resolve each variable definition
     """
+    global_variables = get_global_variables(jinja_env)
     var_deps = variables_dag[var_name]
-    for dep_var_name in var_deps:
+    filtered_var_deps = list(filter(lambda var: var not in global_variables, var_deps))
```

Comment:
wait this shouldn't have happened since global variables are skipped in the dag?

Reply:
As explained below, I left the global function variables in the values so that they will be flattened. (But the global variables are not keys in the dag - since the keys are determined from the raw variables.)